src/main/scala/org/apache/spark/sql/sparkdl_stubs/UDFUtils.scala

```diff
@@ -35,7 +33,7 @@ object UDFUtils extends Logging {
   def registerUDF(sqlCtx: SQLContext, name: String, udf: UserDefinedFunction): UserDefinedFunction = {
     def builder(children: Seq[Expression]) = udf.apply(children.map(cx => new Column(cx)) : _*).expr
     val registry = sqlCtx.sessionState.functionRegistry
-    registry.registerFunction(name, builder)
+    registry.registerFunction(sqlCtx.sessionState.sqlParser.parseFunctionIdentifier(name), builder)
```

Comment:
This line is a mouthful :P but it does seem like the right way to parse UDF names based on https://github.com/apache/spark/pull/17518

Reply:
Thanks for referencing the PR! 

Technically you could pass the identifier straight without calling the sql parse, however I hit an issue with that before - using an id with embedded dashes would fail silently and later you get weird error that the udf is not found so I think it's better/safer to parse it like this.